python/tank/descriptor/io_descriptor/appstore.py

```diff
@@ -538,6 +538,165 @@ def get_latest_cached_version(self, constraint_pattern=None):
+
+            # Check if this version exists locally
+            if not temp_desc.exists_local():
+                log.debug(
+                    f"Version {version_str} not found in cache (expected at {temp_desc.get_path()})"
+                )
+                return (False, None)
+
+            # Check Python compatibility
+            manifest = temp_desc.get_manifest(constants.BUNDLE_METADATA_FILE)
+            if self._check_minimum_python_version(manifest):
+                log.debug(f"Found compatible cached version: {version_str}")
+                return (True, temp_desc)
+
+            # Not compatible - log and return
+            min_py_ver = manifest.get("minimum_python_version", "not specified")
+            log.debug(
+                f"Cached version {version_str} requires Python {min_py_ver} - not compatible"
+            )
+            return (False, None)
+
+        except Exception as e:
+            log.debug(
+                f"Could not check compatibility for cached version {version_str}: {e}"
+            )
+            return (False, None)
+
+    def _find_compatible_cached_version(self, sg_matching_records):
+        """
+        Searches locally cached versions for the highest version compatible
+        with the current Python version.
+
+        :param sg_matching_records: List of Shotgun version records from App Store
+        :returns: IODescriptorAppStore instance or None if no compatible version found
+        """
+        log.debug(
+            "Searching for compatible cached version of %r in bundle cache..." % self
+        )
+
+        # Get all locally cached versions
+        all_versions = self._get_locally_cached_versions()
+        if not all_versions:
+            log.debug("No cached versions found in bundle cache")
+            return None
+
+        log.debug("Found %d cached versions to check" % len(all_versions))
+
+        # Apply filters: label and App Store availability
+        version_numbers = self._filter_versions_by_label(all_versions)
+        if not version_numbers:
+            log.debug("No cached versions match label criteria")
+            return None
+
+        version_numbers = self._filter_versions_by_app_store(
+            version_numbers, sg_matching_records
+        )
+        if not version_numbers:
+            log.debug("No cached versions are available in App Store")
+            return None
+
+        # Sort versions (newest first)
+        sorted_versions = self._sort_versions_descending(version_numbers)
+
+        log.debug(
+            f"Checking {len(sorted_versions)} cached versions for Python compatibility (newest first)"
+        )
+
+        # Find first compatible version
+        for version_str in sorted_versions:
+            is_compatible, descriptor = self._is_version_python_compatible(version_str)
+            if is_compatible:
+                return descriptor
+
+        log.debug("No compatible cached version found")
+        return None
```

Comment:
No need for a return statement at the end of the method. This is Python's default behaviour
```suggestion
```